### PR TITLE
feat(scripts): add MCP server health check script

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

- **check-mcp-servers.sh**: Comprehensive MCP server health checking script
- **common.sh integration**: Reusable `check_mcp_server_health()` function
- **Prerequisite integration**: Health checks in setup scripts

### New Scripts
- `scripts/bash/check-mcp-servers.sh` - Main health check script
- `scripts/bash/test-mcp-health-check.sh` - Test suite (10 test cases)

### Enhanced Scripts
- `scripts/bash/common.sh` - Added health check function
- `scripts/bash/check-prerequisites.sh` - Added MCP health validation
- Other setup scripts - Made executable

## Features

- Parse `.mcp.json` for server configurations
- Validate process running status via `ps`/`pgrep`
- Test actual MCP connectivity
- Color-coded output (✓ green, ✗ red, ⚠ yellow)
- Actionable remediation steps for failures

## Test Plan

- [x] Script parses .mcp.json correctly
- [x] Detects running/stopped MCP processes
- [x] Provides clear remediation steps
- [x] Color-coded output works
- [x] Test suite passes

## Edge Cases Handled

- Missing .mcp.json → Warning, not error
- Empty server list → Skips check
- Network connectivity issues → Clear error message
- Permission issues → Logged with suggestions

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/bash/check-mcp-servers.sh [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        # Check if process is still running
+        if kill -0 "$server_pid" 2>/dev/null; then
+            # Server started successfully
+            echo "success" > "$tmp_file"
+            kill -TERM "$server_pid" 2>/dev/null || true
+        else
+            # Server crashed immediately
+            echo "failed" > "$tmp_file"
+        fi
+    ) &
```
[the remark this answers is not in the document]
Comment (jpoley) [reply scripts/bash/check-mcp-servers.sh]: fixme
